src/main/java/org/opentripplanner/routing/impl/GraphPathFinder.java

```diff
@@ -194,6 +197,14 @@ public GraphPathFinder(Router router) {
                         options.flexMaxCallAndRideSeconds = Math.min(constantLimit, ratioLimit);
                     }
                 }
+
+                Set<String> providersDisallowed = path.states.stream()
+                        .filter(State::isCurrentlyRentingVehicle)
+                        .map(State::getCurrentVehicle)
+                        .map(VehicleDescription::getProvider)
+                        .map(Provider::getProviderName)
+                        .collect(Collectors.toSet());
+                options.vehicleValidator.addFilter(ProviderFilter.providersDisallowedFilter(providersDisallowed));
```

Comment:
Zakazujemy tu providera nawet jeśli provider ma inny typ pojazdów (patrz blinkee) ale zrobienie tego inaczej będzie kłopotliwe plus to trochę corner case. A inny pojazd może wyskoczyć w innym presecie. To nie jest właściwie pytanie, nie wiem czemu to napisałem :D

Reply:
Dlaczego problematyczne? Wymaga napisania tylko oddzielnego filtra, który jest bardzo krótki. Z `VehcileDescription` da się wyciągnąć co to za pojazd.

Reply:
To jest corner case, który dotyczy tylko `hop.city`, wiedziałem o nim już dużo wcześniej, ale nie chciało mi się tego klepać. Główną motywacją było to, że robimy po prostu wiele zapytań. Mogę to dodać jeśli chcecie

Reply:
Omiń go, szansa że na jednej trasie możemy skorzystać z obu pojazdów hopcity